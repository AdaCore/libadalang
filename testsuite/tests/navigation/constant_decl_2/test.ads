package Test is
   I, J : constant Integer;
   --% node.f_ids[0].p_next_part()
   --% node.f_ids[1].p_next_part()

   X : constant Integer;
   --% node.f_ids[0].p_next_part()
   Y : constant Integer;
   --% node.f_ids[0].p_next_part()
private
   I : constant Integer := 42;
   --% node.f_ids[0].p_previous_part()
   J : constant Integer := 42;
   --% node.f_ids[0].p_previous_part()

   X, Y : constant Integer := 42;
   --% node.f_ids[0].p_previous_part()
   --% node.f_ids[1].p_previous_part()
   --% node.f_ids[0].p_canonical_part()
   --% node.f_ids[1].p_canonical_part()
end Test;
