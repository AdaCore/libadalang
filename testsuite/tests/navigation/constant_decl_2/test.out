Working on node <ObjectDecl ["I", "J"] test.ads:2:4-2:28>
=========================================================

Eval 'node.f_ids[0].p_next_part()'
Result: <DefiningName "I" test.ads:11:4-11:5>

Eval 'node.f_ids[1].p_next_part()'
Result: <DefiningName "J" test.ads:13:4-13:5>

Working on node <ObjectDecl ["X"] test.ads:6:4-6:25>
====================================================

Eval 'node.f_ids[0].p_next_part()'
Result: <DefiningName "X" test.ads:16:4-16:5>

Working on node <ObjectDecl ["Y"] test.ads:8:4-8:25>
====================================================

Eval 'node.f_ids[0].p_next_part()'
Result: <DefiningName "Y" test.ads:16:7-16:8>

Working on node <ObjectDecl ["I"] test.ads:11:4-11:31>
======================================================

Eval 'node.f_ids[0].p_previous_part()'
Result: <DefiningName "I" test.ads:2:4-2:5>

Working on node <ObjectDecl ["J"] test.ads:13:4-13:31>
======================================================

Eval 'node.f_ids[0].p_previous_part()'
Result: <DefiningName "J" test.ads:2:7-2:8>

Working on node <ObjectDecl ["X", "Y"] test.ads:16:4-16:34>
===========================================================

Eval 'node.f_ids[0].p_previous_part()'
Result: <DefiningName "X" test.ads:6:4-6:5>

Eval 'node.f_ids[1].p_previous_part()'
Result: <DefiningName "Y" test.ads:8:4-8:5>

Eval 'node.f_ids[0].p_canonical_part()'
Result: <DefiningName "X" test.ads:6:4-6:5>

Eval 'node.f_ids[1].p_canonical_part()'
Result: <DefiningName "Y" test.ads:8:4-8:5>
